src/main/java/anticope/rejects/modules/AutoEnchant.java

```diff
@@ -0,0 +1,130 @@
+            .name("delay")
+            .description("The tick delay between enchanting items.")
+            .defaultValue(50)
+            .sliderMax(500)
+            .min(0)
+            .build()
+    );
+
+    private final Setting<Integer> level = sgGeneral.add(new IntSetting.Builder()
+            .name("level")
+            .description("Choose enchantment levels 1-3")
+            .defaultValue(3)
+            .max(3)
+            .min(1)
+            .build()
+    );
+
+    private final Setting<Boolean> drop = sgGeneral.add(new BoolSetting.Builder()
+            .name("drop")
+            .description("Automatically drops enchanted items (useful for when not enough inventory space)")
+            .defaultValue(false)
+            .build()
+    );
+
+    private final Setting<List<Item>> itemWhitelist = sgGeneral.add(new ItemListSetting.Builder()
+            .name("item-whitelist")
+            .description("Item that require enchantment.")
+            .defaultValue()
+            .filter(item -> item.equals(Items.BOOK) || item.isDamageable())
+            .build()
+    );
+
+    public AutoEnchant() {
+        super(MeteorRejectsAddon.CATEGORY, "auto-enchant ", "Automatically enchanting items.");
+    }
+
+    @EventHandler
+    private void onOpenScreen(OpenScreenEvent event) {
+        if (!(Objects.requireNonNull(mc.player).currentScreenHandler instanceof EnchantmentScreenHandler))
+            return;
+        MeteorExecutor.execute(this::autoEnchant);
+    }
+
+    private void autoEnchant() {
+        if (!(Objects.requireNonNull(mc.player).currentScreenHandler instanceof EnchantmentScreenHandler handler))
+            return;
+        if (mc.player.experienceLevel < 30) {
+            info("You don't have enough experience levels");
+            return;
+        }
+        while (getEmptySlotCount(handler) > 2 || drop.get()) {
+            if (!(mc.player.currentScreenHandler instanceof EnchantmentScreenHandler)) {
+                info("Enchanting table is closed.");
+                break;
+            }
+            if (handler.getLapisCount() < 3 && !fillLapisItem()) {
+                info("Lapis lazuli is not found.");
+                break;
+            }
+            if (!fillCanEnchantItem()) {
+                info("No items found to enchant.");
+                break;
+            }
+            Objects.requireNonNull(mc.interactionManager).clickButton(handler.syncId, level.get() - 1);
+            if (getEmptySlotCount(handler) > 2) {
+                InvUtils.shiftClick().slotId(0);
+            } else if (drop.get() && handler.getSlot(0).hasStack()) {
+                // I don't know why an exception LegacyRandomSource is thrown here,
+                // so I used the main thread to drop items.
+                mc.execute(() -> InvUtils.drop().slotId(0));
+            }
+
+            // Why sleep here? I don't know either.
+            try {
+                Thread.sleep(delay.get());
```

Comment:
this is the exact use case for a java `ScheduledExecutorService`

Reply:
Idea provided suggestions for my code, indicating that this is an problem, and perhaps we can implement it more elegantly.